### PR TITLE
[Apple] Support Mac Catalyst

### PR DESCRIPTION
**Pull request by DmitriyKirakosyan**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

This PR adds support for Mac Catalyst by adding the requiring binaries to the `AppCenterReactNativeShared` 
 xcfromework.

## Related PRs or issues

This addresses the issue https://github.com/microsoft/appcenter-sdk-react-native/issues/901

## Misc

Fat framework was converted to xcframework here https://github.com/microsoft/appcenter-sdk-react-native/pull/896 for the `AppCenterReactNativeShared` project, however the mac catalyst binary was missing in the final xcframework.

### CODE REVIEW

**Review by MatkovIvan — CHANGES_REQUESTED** [review]
```
AppCenterReactNativeShared.xcframework/ios-x86_64-maccatalyst/AppCenterReactNativeShared.framework/Headers/Headers: No such file or directory
```

**Review by MatkovIvan — APPROVED** [review]

**Review by russelarms — CHANGES_REQUESTED** [review]
TestApp build failed on Catalyst (Big Sur) with:
 ```
AppCenterReactNativeAnalytics/AppCenterReactNativeAnalytics.m:28:9: 'AppCenterReactNativeShared/AppCenterReactNativeShared.h' file not found
```
on Catalina:
```
Undefined symbols for architecture x86_64
```

**Review by russelarms — COMMENTED** [review]
Errors were due to Cocoapods version 1.10.0. An update to 1.10.1 resolved them.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #### iOS
 
 * **[Feature]** Use XCFramework format for the AppCenterReactNativeShared pod, it allows running the SDK on Apple Silicon simulators.
+* **[Feature]** Support Mac Catalyst.
```
Comment (DmitriyKirakosyan) [inline CHANGELOG.md:10]:
```suggestion
* **[Feature]** Support Mac Catalyst. Cocoapods 1.10.1+ is required.
```

**Review by MatkovIvan — DISMISSED** [review]

**Review by russelarms — APPROVED** [review]

**Pull request merged**